### PR TITLE
Creates Test Plan for Running Integration Tests on Server API Endpoints

### LINKED ISSUE #1400: Engineer tests and validation for 'Account' Web API requests
### Overview

We need to create a workflow for testing all 'account' endpoints. We will need to set up tests and web API request validation for the endpoints.

### Details | Scope of Work
We need to test the API endpoints using 'jest'  

### Action Items

- [x] write all tests for endpoints and documentation
- [x] set up server side validation schema

### Resources/Instructions
Server Side Validation
- https://www.npmjs.com/package/express-json-validator-middleware
- https://simonplend.com/how-to-handle-request-validation-in-your-express-api/  

Testing
- https://jestjs.io/

### PR DESCRIPTION
**Pull request by agosmou**
Fixes #1400 

### What changes did you make?

- Added Test Plan for use in running integration tests on the TDM Server:
    - Added 'Jest' to run tests
    - Implemented node library 'testcontainers' to create a docker container for an instance of a test MS SQL Server database
    - Added the integration tests to the TDM CI pipeline so they run on every pull request to the 'develop' branch
- Added integration tests for the 'accounts' endpoints that test the functional requirements of the user managemnet workflow as well as edge cases
- Updated the [Test Wiki](https://github.com/hackforla/tdm-calculator/wiki/Testing#testing-the-tdm-server) to include documentation on the new Test Plan and instructions to add future tests

### Why did you make the changes (we will use this info to test)?

- **Reliable Testing**: Integrated 'Jest' and 'testcontainers' to add robust testing reflecting our production setup
- **Enhanced CI/CD**: Integration tests in TDM CI pipeline prevent PRs that merge breaking changes
- **User Workflow Quality**: Added tests for 'accounts' endpoints to ensure user management workflow meets functional requirements and handles edge cases.

### References
**Jest**
- [Jest Documentation](https://jestjs.io/docs/en/getting-started)
- Setting up files to be used [before each test](https://stackoverflow.com/questions/49685265/jest-mock-user-module-in-all-test-files)
- [Jest Config](https://jestjs.io/docs/configuration#setupfiles-array)

**Supertest**
- [Supertest Documentation](https://www.npmjs.com/package/supertest)

**Testcontainers**
- [Testcontainers Documentation](https://testcontainers.com/)
- Testcontainers [for node](https://node.testcontainers.org/features/containers/)
- I had to handroll our own implementation, but there is a formal module [coming out soon](https://github.com/testcontainers/testcontainers-node/pull/645) that I used as reference

**SendGrid** (Mocked for Testing)
- [Sendgrid API docs](https://docs.sendgrid.com/for-developers/sending-email/quickstart-nodejs)
- Sendgrid mocked using [jest](https://jestjs.io/docs/mock-function-api)

**Flyway Migrations**
- [Source Code](https://github.com/flyway/flyway/blob/5e894fed50b9ebdb122b1f6671679cb69e17e84c/flyway-core/src/main/java/org/flywaydb/core/internal/configuration/ConfigUtils.java#L117) observed to use in flyway config to disable reports when testing

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -75,6 +75,8 @@ app.use((err, req, res) => {
 
 app.use(errorHandler);
 
-app.listen(port, () => console.log(`Server running on port ${port}`));
+if (process.env.TEST_ENV !== "true") {
```
Comment (agosmou) [inline server/server.js:78]: Added a minor conditional to be able to instantiate the server, i.e. set them up and tear them down. Development environments and production environments will have a falsy conditional; thus, be unaffected.

this was a necessary change to be able run multiple test suites with a real server.

**Review by agosmou — COMMENTED** [review]
Added comments to files to explain

**Comment by agosmou** [comment]
I tried to keep commits atomic, but there was **A LOT** of refactoring; hence, the large amount of commits. We can squash and merge to condense this.

**Comment by agosmou** [comment]
[Wiki](https://github.com/hackforla/tdm-calculator/wiki/Testing#testing-the-tdm-server) is linked on comment above. Do review for further info.

**Comment by agosmou** [comment]
@entrotech 
I shared the .env for this file on the shared drive so you can use it in your local environment when reviewing.

alternatively you can grab the env variables from the github actions file - just make sure all other variables in your .env are commented out!

**Comment by agosmou** [comment]
@entrotech 

The below should facilitate interpretation of code

***

**Suggested Order for File Review (root directory would be server):**

_Environment Variables_
0. `.example.env`

_Setting up a containerized db with application code_
1a. `utils/mssql-container-setup.js`
1b. `db/flyway-config.js`
1c. `global-setup/global-setup.js`
1d. `global-setup/global-teardown.js`

_Server refactors and setup for use as an instance on tests_
2a. `server.js`
2b. `utils/server-setup.js`
2c. `local-setup/setup-after-env.js`

_Testing configs and script_
3a. `jest.config.js`
3b. `package.json`

_Accounts tests_
4. `__tests__/account.test.js`

_Setting up github actions_
5. `.github/workflows/tdm-server-tests.yml`
